puppet-runner.go

```diff
@@ -150,7 +151,12 @@ func main() {
 	}
 
 	if environment == "" || !isValidEnvironment(environment) {
-		log.Printf("Environment %q is invalid; resetting", environment)
+		if *envResetDelay != 0 {
+			log.Printf("Environment %q is invalid, waiting %v minutes to reset and retry", environment, envResetDelay)
+			time.Sleep(time.Duration(*envResetDelay) * time.Minute)
```

Comment:
I'm not familiar with this code but a question. While we are sleeping for say 1 hour, can that code block be called again for the same node? In other words, is this trigerred externally multiple times while on the feature environment? Because that could queue multiple thread wait and switch.

Reply:
There is a lock that gets generated as best I recall that should make it so only one instance can run at a time. My main concern is that this will cause nodes to show in PE as simply not reporting in x amount of time instead of showing them as failed runs due to no such env. I am stuck on the latter because it is clear in the console what is happening.

Reply:
Am I being overly picky here? @yachub - any specific thoughts?

Reply:
When agents request an invalid environment, they don't submit a report. So in that case, you would also see the agent in the non-reporting category.

Reply:
Testing locally it looks like I can trigger two processes running at the same time, so unfortunately there doesn't seem to be any lock in place.

Reply:
Oh yea... that was a casualty of making it cross platform plus puppet locks so only one actual agent run would ever happen.

I think when an agent tries to use a missing env something shows up in PE like when there is a compile error... I may be wrong though

Reply:
Puppet's own locking system seems to work.  I can call two instances of the runner, but puppet will exit if there's already another agent run in progress
```
2021/07/16 13:28:41 Running 'puppet agent --no-daemonize --onetime'
2021/07/16 13:28:43 exit status 1
```
Do we want to merge this then?  Confirmed that the missing environment doesn't result in any report or anything visible in the console.